src/components/Widget/components/Conversation/components/Sender/index.tsx

```diff
@@ -14,41 +17,123 @@ type Props = {
   useEffect(() => { if (showChat && autofocus) inputRef.current?.focus(); }, [showChat]);
+  useEffect(() => { setFirefox(isFirefox())}, [])
+
+  useImperativeHandle(ref, () => {
+    return {
+      onSelectEmoji: handlerOnSelectEmoji,
+    };
+  });
 
   const handlerOnChange = (event) => {
     onTextInputChange && onTextInputChange(event)
   }
 
   const handlerSendMessage = () => {
-    const { current } = inputRef
-    if(current?.innerHTML) {
-      sendMessage(current.innerText);
-      current.innerHTML = ''
+    const el = inputRef.current;
+    if(el.innerHTML) {
+      sendMessage(el.innerText);
+      el.innerHTML = ''
     }
   }
 
+  const handlerOnSelectEmoji = (emoji) => {
+    const el = inputRef.current;
+    const { start, end } = getSelection(el)
+    if(el.innerHTML) {
+      const firstPart = el.innerHTML.substring(0, start);
+      const secondPart = el.innerHTML.substring(end);
+      el.innerHTML = (`${firstPart}${emoji.native}${secondPart}`)
+    } else {
+      el.innerHTML = emoji.native
+    }
+    updateCaret(el, start, emoji.native.length)
+  }
+
   const handlerOnKeyPress = (event) => {
+    const el = inputRef.current;
+
     if(event.charCode == 13 && !event.shiftKey) {
       event.preventDefault()
       handlerSendMessage();
     }
+    if(event.charCode === 13 && event.shiftKey) {
+      event.preventDefault()
+      insertNodeAtCaret(el);
+      setEnter(true)
+    }
+  }
+
+  // TODO use a context for checkSize and toggle picker
+  const checkSize = () => {
+    const senderEl = refContainer.current
+    if(senderEl && height !== senderEl.clientHeight) {
+      const {clientHeight} = senderEl;
+      setHeight(clientHeight)
+      onChangeSize(clientHeight ? clientHeight -1 : 0)
+    }
+  }
+
+  const handlerOnKeyUp = (event) => {
+    const el = inputRef.current;
+    if(!el) return true;
+    // Conditions need for firefox
+    if(firefox && event.key === 'Backspace') {
+      if(el.innerHTML.length === 1 && enter) {
+        el.innerHTML = ''
+        setEnter(false)
+      }
+      else if(brRegex.test(el.innerHTML)){
+        el.innerHTML = el.innerHTML.replace(brRegex, '')
+      }
+    }
+    checkSize()
+  }
+
+  const handlerOnKeyDown= (event) => {
+    const el = inputRef.current;
+    
+    if( event.key === 'Backspace' && el){
+      const caretPosition = getCaretIndex(inputRef.current)
+      const character = el.innerHTML.charAt(caretPosition - 1)
+      if(character === "\n") {
+        event.preventDefault();
+        event.stopPropagation()
+        el.innerHTML = (el.innerHTML.substring(0, caretPosition - 1) + el.innerHTML.substring(caretPosition))
+        updateCaret(el, caretPosition, -1)
+      }
+    }
+  }
+
+  const handlerPressEmoji = () => {
+    onPressEmoji();
+    checkSize();
   }
 
   return (
-    <div className="rcw-sender">
+    <div ref={refContainer} className="rcw-sender">
+      <button className='rcw-picker-btn' type="submit" onClick={handlerPressEmoji}>
```

Comment:
I think we should add a conditional to hide the emoji picker setting it to true by default. What do you think?